src/framework/theme/components/select/select.component.ts

```diff
@@ -422,7 +434,7 @@ export class NbSelectComponent<T> implements OnInit, AfterViewInit, AfterContent
   protected handleSingleSelect(option: NbOptionComponent<T>) {
     const selected = this.selectionModel.pop();
 
-    if (selected && selected !== option) {
+    if (selected && !this.compareWith(selected, option)) {
```

Comment:
`selected` and `option` is instances of NbOptionComponent component. We can't compare them using possibly user defined comparator, since it expects to compare option values which type is `T` not `NbOptionComponent`.
Please, pass options values to comparator here.